clang/lib/Lex/Preprocessor.cpp

```diff
@@ -1551,6 +1567,58 @@ bool Preprocessor::isPPInSafeBufferOptOutRegion(SourceLocation &StartLoc) {
   return InSafeBufferOptOutRegion;
 }
 
+SmallVector<SourceLocation, 64>
+Preprocessor::serializeSafeBufferOptOutMap() const {
+  assert(!InSafeBufferOptOutRegion &&
```

Comment:
We require the pragma to be closed before the end of TU right? And that's a hard error, not a warning?

Reply:
It's a warning.

Reply:
Yeah in this case we need to support the situation when the warning was ignored.

Reply:
Oh,  I'm sorry,  they are not warnings, they are errors: https://github.com/llvm/llvm-project/blob/c79690040acf5bb3d857558b0878db47f7f23dc3/clang/lib/Lex/Pragma.cpp#L1265